receiver/awsxrayreceiver/internal/socketconn/socketconn.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Copyright 2018-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the License. A copy of the License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+package socketconn
+
+// SocketConn is an interface for socket connection.
+type SocketConn interface {
```

Comment:
I suppose this is used to help with mocking in tests, can it be defined on the tests instead?

Reply:
I don't think this is possible because this interface is used in both actual code (i.e. `poller.go`) and tests. And I can't move all the content of `poller.go` to be `poller_test.go`. Or maybe I misunderstood you?

Reply:
Use the original type in the actual code and cast to the convenience type only tests. It will require a cast somewhere on the tests (you can wrap it in a single function) but it avoids a construct that is intended for tests to be on production code - up to you.

Reply:
Thanks for clarifying. Sorry for my ignorance but I'm still a bit confused. Are you suggesting doing things like this:

```
type poller struct {
	udpSock    *net.UDPConn
        ......
}
```

And in tests, I can define `mockSocketConn` as follows and initialize `poller` with it:
```
type mockSocketConn struct {
......
}

p := &poller{
    udpSock: &mockSocketConn{......},
}

// In some tests, cast udpSock to the mockSocketConn

p.udpSock.(*mockSocketConn)......
```

Since `net.UDPConn` is a concrete type not an interface, I thought the `poller` initialization code above will not compile. And the only way I know to make it possible for the `udpSock` field to be able to hold instances of multiple concrete type is to define an interface, which is the `SocketConn` here. Could you clarify?

Thanks!

Reply:
My mistake: I was thinking that the code was dealing with an interface, `net.PacketConn`, not a concrete type.